examples/image-telephone/application.py

```diff
@@ -0,0 +1,170 @@
+"""
+This module demonstrates a telephone application
+using Burr that:
+ - captions an image
+ - creates caption embeddings (for analysis)
+ - creates a new image based on the created caption
+"""
+import os
+import uuid
+
+import requests
+from hamilton import dataflows, driver
+
+from burr.core import Action, ApplicationBuilder, State, default, expr
+from burr.core.action import action
+from burr.lifecycle import PostRunStepHook
+
+# import hamilton modules
+caption_images = dataflows.import_module("caption_images", "elijahbenizzy")
+generate_images = dataflows.import_module("generate_images", "elijahbenizzy")
+
+
+class ImageSaverHook(PostRunStepHook):
```

Comment:
Hmm I wonder if this should be a persister? It can save the image and load up the image URL?